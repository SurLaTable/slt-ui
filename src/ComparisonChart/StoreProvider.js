import React from 'react';
// import { render } from 'react-dom';
import { Provider } from 'react-redux';
import { createStore } from 'redux';

import sltReducers from './reducers/sltReducers';

const sltStore = createStore(
    sltReducers,
    // Enable DevTools, switch to localStorage on Prod:
    window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
);

const StoreProvider = (props) => (
    <Provider store={sltStore}>
			<React.Fragment>{props.children}</React.Fragment>
    </Provider>
);
export default StoreProvider;
