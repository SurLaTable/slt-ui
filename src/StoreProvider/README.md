```
<div data-component="StoreProvider"></div>
```
