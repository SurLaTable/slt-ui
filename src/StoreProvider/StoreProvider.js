import React from 'react';
import PropTypes from 'prop-types';
import { Provider } from 'react-redux';
import configureStore from '../utils/storeTools';

export const store = configureStore();

const StoreProvider = (props) => (
	<Provider store={store}>
		<React.Fragment>{props.children}</React.Fragment>
	</Provider>
);

StoreProvider.propTypes = {
	children: PropTypes.node
};

export default StoreProvider;
