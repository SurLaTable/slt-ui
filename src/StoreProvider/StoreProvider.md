```jsx
<StoreProvider />
```

```html
<div data-component="StoreProvider"></div>
```
