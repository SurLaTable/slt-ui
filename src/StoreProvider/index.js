import React from 'react';
import { Provider } from 'react-redux';
import configureStore from '../utils/storeTools';

export const store = configureStore();
const StoreProvider = (props) => (
	<Provider store={store}>
		<React.Fragment>{props.children}</React.Fragment>
	</Provider>
);
export default StoreProvider;
