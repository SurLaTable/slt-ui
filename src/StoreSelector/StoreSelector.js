import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withStyles } from '@material-ui/core/styles';
import withWidth, { isWidthDown } from '@material-ui/core/withWidth';

import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogTitle from '@material-ui/core/DialogTitle';
import DialogContent from '@material-ui/core/DialogContent';
import DialogActions from '@material-ui/core/DialogActions';

import StoreList from '../StoreList';
import LocationField from '../LocationField';
import LocationButton from '../LocationButton';
import ArrowUpwardIcon from '@material-ui/icons/ArrowUpward';
import CloseIcon from '@material-ui/icons/close';
import Zoom from '@material-ui/core/Zoom';

import * as sltStoresApi from '../services/slt-stores';
import { selectors as googleMapsSelectors } from '../services/google-maps';

function throttle(fn, wait) {
	var time = Date.now();
	var debounce;
	return function() {
		var delta = time + wait - Date.now();
		debounce = clearTimeout(debounce);
		if (delta < 0) {
			fn.apply(this, arguments);
			time = Date.now();
		} else {
			debounce = setTimeout(() => {
				fn.apply(this, arguments);
				time = Date.now();
			}, delta);
		}
	};
}

const style = (theme) => {
	return {
		flexContainer: {
			display: 'flex',
			alignItems: 'center'
		},
		flex: {
			flex: 1
		},
		display: {
			display: 'flex',
			flexDirection: 'column',
			border: `1px solid ${theme.palette.grey[400]}`,
			borderRadius: 0,
			padding: '8px 8px',
			justifyContent: 'space-evenly',
			background: `${theme.palette.grey[100]}`
		},
		button: {
			borderRadius: 0,
			minHeight: theme.typography.pxToRem(theme.typography.fontSize),
			lineHeight: theme.typography.pxToRem(theme.typography.fontSize),
			boxShadow: theme.shadows[0],
			'&:active': {
				boxShadow: theme.shadows[0]
			}
		},
		fab: {
			position: 'absolute',
			bottom: `calc(100% + ${theme.spacing.unit * 2}px)`,
			right: theme.spacing.unit * 2
		}
	};
};

class StoreSelector extends React.Component {
	static contextTypes = {
		store: PropTypes.object
	};

	constructor() {
		super();
		this.state = {
			open: false,
			showScrollToTop: false,
			scrollingToTop: false,
			scrollTop: 0
		};
		this.animateScrollRequest = null;

		this.handleGoToTop = this.handleGoToTop.bind(this);
		this.toggleOpen = this.toggleOpen.bind(this);
		this.currentScrollElement = null;

		let throttledScroll = throttle((e) => {
			let { scrollTop } = e.target;
			if (scrollTop != this.state.scrollTop) {
				this.setState({ scrollTop });
			}
		}, 300);

		this.handleScroll = (e) => {
			this.currentScrollElement = e.target;

			let { scrollTop } = e.target;
			if (this.animateScrollRequest) {
				if (scrollTop < 2) {
					global.cancelAnimationFrame(this.animateScrollRequest);
					this.animateScrollRequest = null;
					e.target.scrollTop = 0;
					this.setState({ scrollTop: 0 });
				}
			} else {
				e.persist();
				throttledScroll(e);
			}
		};
	}

	componentDidMount() {
		if (this.props.hasLocations == false) {
			this.setState({ open: true });
		}
	}

	toggleOpen() {
		this.setState({ open: !this.state.open });
	}

	handleGoToTop() {
		this.currentScrollElement.scrollTop += (0 - this.currentScrollElement.scrollTop) * 0.3;
		this.animateScrollRequest = global.requestAnimationFrame(this.handleGoToTop);
	}

	render() {
		let { open, scrollTop, showScrollToTop } = this.state;
		let { selectedStore, classes, width, culinary, storeListProps } = this.props;

		const dialog = (
			<Dialog
				fullWidth
				fullScreen={isWidthDown('sm', width)}
				maxWidth={'sm'}
				open={open}
				onClose={this.toggleOpen}
				onScroll={this.handleScroll}
			>
				<DialogTitle>
					<div className={classes.flexContainer}>
						<div className={classes.flex}>Find A Location Near You</div>
						<Button
							variant="flat"
							size="small"
							onClick={this.toggleOpen}
							color="default"
						>
							<CloseIcon />
						</Button>
					</div>
				</DialogTitle>

				<DialogContent>
					<LocationButton>Use My Location</LocationButton>
					<div>
						<LocationField />
					</div>

					<StoreList
						sortBy={'distance'}
						detailed={false}
						onItemSelected={this.toggleOpen}
						onShowMore={(page) => {
							this.setState({ showScrollToTop: page > 1 });
						}}
						culinary={culinary}
						{...storeListProps}
					/>
				</DialogContent>
				<DialogActions>
					<div style={{ position: 'relative' }}>
						<Zoom
							in={showScrollToTop && scrollTop > 600}
							unmountOnExit
						>
							<Button
								variant="fab"
								className={classes.fab}
								color="primary"
								onClick={this.handleGoToTop}
							>
								<ArrowUpwardIcon />
							</Button>
						</Zoom>
					</div>
				</DialogActions>
			</Dialog>
		);

		let display;

		if (selectedStore == null) {
			display = (
				<Paper
					elevation={0}
					className={classes.display}
				>
					<Typography>Class Location:</Typography>
					<Typography>&nbsp;</Typography>
					<Button
						fullWidth
						onClick={this.toggleOpen}
						className={classes.button}
						style={{ flex: 1 }}
					>
						Select Location
					</Button>
				</Paper>
			);
		} else {
			const location = selectedStore.location;
			display = (
				<Paper
					elevation={0}
					className={classes.display}
				>
					<Typography>Class Location:</Typography>
					<Typography>
						<strong>
							{location.city}
							{', '}
							{location.state}
						</strong>
					</Typography>
					<Button
						onClick={this.toggleOpen}
						className={classes.button}
					>
						Select Location
					</Button>
				</Paper>
			);
		}

		return (
			<>
				{display}
				{dialog}
			</>
		);
	}
}

StoreSelector.propTypes = {
	selectedStore: PropTypes.object,
	storeListProps: PropTypes.object,
	culinary: PropTypes.bool
};
StoreSelector.defaultProps = {
	culinary: false
};

const mapStateToProps = (state, props) => {
	return {
		...props,
		hasLocations: googleMapsSelectors.getHasLocations(state),
		selectedStore: sltStoresApi.selectors.getSelectedItemObject(state)
	};
};

export default connect(mapStateToProps)(withWidth()(withStyles(style)(StoreSelector)));
