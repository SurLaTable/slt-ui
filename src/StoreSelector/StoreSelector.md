```jsx
<ThemeProvider>
	<StoreProvider>
		<StoreSelector />
	</StoreProvider>
</ThemeProvider>
```

```html
<div data-component="StoreSelector"></div>
```
