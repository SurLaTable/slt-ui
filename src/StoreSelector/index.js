export { default } from './StoreSelector.js';
