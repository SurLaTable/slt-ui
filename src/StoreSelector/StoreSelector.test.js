import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import StoreProvider from '../StoreProvider';
import StoreSelector from './StoreSelector';

test('Render', () => {
	const renderer = new ShallowRenderer();
	renderer.render(
		<StoreProvider>
			<StoreSelector />
		</StoreProvider>
	);
	const result = renderer.getRenderOutput();
	expect(result).toMatchSnapshot();
});
