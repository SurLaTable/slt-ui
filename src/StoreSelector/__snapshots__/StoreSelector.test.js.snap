// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render 1`] = `
<Provider
  store={
    Object {
      "asyncReducers": Object {
        "C:\\\\Users\\\\dmeagher\\\\Desktop\\\\slt-ui\\\\src\\\\services\\\\google-maps": [Function],
        "C:\\\\Users\\\\dmeagher\\\\Desktop\\\\slt-ui\\\\src\\\\services\\\\slt-stores": [Function],
      },
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <React.Fragment>
    <Connect(WithTheme(WithWidth(WithStyles(StoreSelector)))) />
  </React.Fragment>
</Provider>
`;
